mlflow/tensorflow.py

```diff
@@ -471,13 +471,23 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
     def on_train_begin(self, logs=None):  # pylint: disable=unused-argument
         opt = self.model.optimizer
-        if hasattr(opt, 'optimizer'):
+        # Secondary checks are if the optimizer is a TensorFlow optimizer rather than a Keras one.
```

Comment:
Can we pull the reassignment of `opt` up into its own if-statement before logging any params & explain why we do it? e.g.

```
# Explain why we do this
if hasattr(opt, 'optimizer'):
    opt = opt.optimizer
# Log optimizer attributes...
if hasattr(opt, '_name'):
```

Or if there's a reason we bundle assigning `opt` & logging the `name` together in the same branch, could we add a clarifying comment (e.g. because we want to log the name of the outer optimizer, but extract the inner optimizer so that we can log its learning rate & epislon)? Or maybe just log the name first, and then reassign `opt` afterwards? I think either would help make this clearer